### DISCUSSION TITLE
Trying to load .wasm file from browser cache storage

### DISCUSSION THREAD

**Discussion by kishorekaruppusamy** [microsoft/onnxruntime#19536] [Other Q&A]
I am woking on onnx model for the past 2 months, as of now the .wasm file loading from disc cache for every time i am creating a session using my onnx model check the image below ..
![image](https://github.com/microsoft/onnxruntime/assets/103121724/1e37f100-df3b-4814-8499-da0f87f98a64)

it take 80 ms to load this file from disc cache every time. 
This significantly impact my inference time, so i am trying to reduce this time 
For this i have a idea that creating webassembly instance using arraybuffer of .wasm file so the instead of using this .wasm file disc cache i can load .wasm file as arraybuffer from indexed db ..

As of now i checked this wasm files are loaded internally by ort file, is there any way to make .wasm files load using arraybuffer instead of loading from disc cache .. ( suggession are most welcome )

**Comment by fs-eire** [comment]
Could you try to create an object URL from the blob that you get from indexed DB, and set that URL to `ort.env.wasm.wasmPaths['ort-wasm-simd.wasm']`? ([API reference link](https://onnxruntime.ai/docs/api/js/interfaces/Env.WebAssemblyFlags.html#wasmPaths)) please let me know if it works.

I am also curious about why 80ms is considered "signigicant impact" to your inference time. I am not sure what is your use case but in my understanding, 80ms is totally OK for a web app for initialization. There going to be much more time consuming operations...
